### ISSUE TITLE
Something wrong in PreprocessMicrographs in devel

### ISSUE THREAD

**Issue by dmaluenda** [I2PC/scipion#1437]
6 Fails appear with `scipion test em.packages.xmipp3.tests.test_protocols_xmipp_mics.TestXmippExtractParticles`

It is not generating output. In one subTest (same as picking) it generates an incorrect output. It seems that is related with the steps creation.

I realized that the problem appears only if there are a 'preprocess micropgaphs' protocol in the workflow (with a downSampling).

@rsanchezgarc has reported the same problem when concatenates the 'preprocess micrographs' with 'Xmipp manual picking'. Thus, we thing that is someting related with the preprocess...

Any idea @cossorzano @pconesa @majtner ??

**Comment by majtner** [comment]
I am not sure if the problem is in preprocessing micrographs. If you look, for example, at TestXmippAutomaticPicking (which is in same test file), it is using the same downsampling with factor 5 and it is working ok. 
Also TestXmippPreprocessMicrographs is working ok (except for some error in normalization, which is not related with downsampling).

In my opinion, the problem seems to be in test TestXmippExtractParticles. For example, why it is performing downsampling for "faked picking" but subsequent tests are using faked picking coordinates and original images?

**Comment by rsanchezgarc** [comment]
In my case, I did not used ExtractParticles. Just PreprocessMicrographs
followed by particle picking (gautomatch and xmipp independently) and just
the last micrograph was picked. On the other hand, when I did particke
picking from raw micrographs, the algorithm worked well.


Ruben


2018-02-01 11:00 GMT+01:00 Tomas <notifications@github.com>:

**Comment by majtner** [comment]
Thank you, @rsanchezgarc for these details, I think that I have found it.

It actually seems that there are multiple problems here. When I was comparing the raw micrographs with downsampled ones, there was missing _micName property in downsampled ones. After adding this property, it seems that all mics are used, not only the last one (at least in my test it was working).

However, there is another problem that TestXmippExtractParticles is testing it incorrectly. This test is not producing any output (btw. while finishing without error) and I believe it is because raw set of mics is used instead of the downsampled ones. After changing it, I had the correct output.
I will try to correct it all and prepare a pull request.

**Comment by josuegbl** [comment]
@majtner, I think that TestXmippExtractParticles test is ok. This test check several conditions that can occurs in a real data processing such as, pick from preprocessed microgrpahs and extract from imported (or aligned) ones. In any case, this test runs correctly in V1.2 beta (facilities devel), in which the protocol extract particles is in streaming. I strongly disagree with change the test just because is failing, presumably, due to the changes introduced in preprocess micrographs to support streaming.

**Comment by majtner** [comment]
@josuegbl You are right, TestXmippExtractParticles was correct. I was not changing it because it was failing but because I thought it was computing something incorrect (I was asking about it in my first comment).

Anyway, I revert the changes in TestXmippExtractParticles, I only removed micsId from testExtractNoise because it is not used there and changed downsampling factor to be the same as in preprocessing (testExtractNoise was added later and it is not in V1.2 beta (facilities devel)).

The problem in preprocess micrographs was with incorrect paths to processed micrographs. I corrected it and now all tests are passing. 

Thank you for spotting this error.
